lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -263,12 +278,19 @@ def configure(conf)
       # Functions.
       @running_cloudfunctions = false
 
-      # Set labels, etc. based on the config
+      # Set up the MonitoredResource, labels, etc. based on the config
       case @platform
       when Platform::GCE
-        @service_name = COMPUTE_SERVICE
+        @resource.type = COMPUTE_RESOURCE_TYPE
+        # TODO: introduce a new MonitoredResource-centric configuration and
+        # deprecate subservice-name; for now, translate known uses.
         if @subservice_name
-          @service_name = @subservice_name
+          # TODO: what should we do if we encounter an unknown value?
+          if @subservice_name == DATAFLOW_SERVICE
+            @resource.type = DATAFLOW_RESOURCE_TYPE
+          elsif @subservice_name == ML_SERVICE
+            @resource.type = ML_RESOURCE_TYPE
```

Comment:
Given that users can take a long time to update their fluentd on various platforms, and the v1 logging API turndown will be a significant motivation to update, I think it makes better sense to get it right on the first move to v2.